src/_guides/language/sound-dart.md

```diff
@@ -1,54 +1,51 @@
 ---
-title: Strong Mode Dart
+title: Dart's Type System
 description: Why and how to write sound Dart code.
 ---
 <?code-excerpt replace="/([A-Z]\w*)\d\b/$1/g; /\b(main)\d\b/$1/g"?>
 
-{% comment %}
-update-for-dart-2
-{% endcomment %}
+Dart has a sound static
+type system that uses a combination of static and [runtime checks](#runtime-checks) to
+ensure that your code is type safe: runtime types always match static types.
+Because Dart has [type inference](#type-inference),
+you don't always have to specify types.
 
-This guide tells you why and how to write sound (type safe) Dart code.
-You'll learn how to use strong mode to enable soundness, as well as
-how to substitute types safely when overriding methods.
-
-**Strong mode** is a sound static
-type system that uses a combination of static and runtime checks to
-ensure your code is type safe&mdash;that you can never see a value
-whose runtime type does not match its static type.
-With strong mode enabled (in an implementation that has both the
-static and runtime checks), Dart is a sound language.
-
-<aside class="alert alert-warning" markdown="1">
-  **Warning:**
-  Currently, the Dart dev compiler
-  ([dartdevc,]({{site.webdev}}/tools/dartdevc) also known as _DDC_)
-  is the only full implementation of strong mode.
-  VM and dart2js support are on their way.
-</aside>
+This page concentrates on the type safety features added in Dart 2.
+For a full introduction to the Dart language, including types, see the
+[language tour](/guides/language/language-tour).
 
 <aside class="alert alert-info" markdown="1">
   **Terminology note:**
-  The terms **sound** Dart, **strong mode** Dart, and **type safe** Dart
-  are sometimes used interchangeably.
-  **Classic** Dart refers to Dart before soundness was added to the language.
+  The terms **sound** Dart and **type safe** Dart
+  are often used interchangeably.
+  You might also see the term **strong mode**.
+  Strong mode was an opt-in Dart 1.x feature
+  that provided partial support for type safety.
 </aside>
 
-By writing sound Dart code today, you'll reap some benefits now,
-with more in the near future. Current benefits include finding bugs
-at compile time (rather than at runtime) using Dart's static analyzer.
-And soon you'll be able to use new tools that quickly and incrementally
-compile your sound Dart code, giving you a better overall
-developer experience.
+One benefit of type safety is the ability to find bugs
+at compile time (rather than at runtime) using Dart's [static analyzer.][analyzer]
+Another benefit is being able to quickly and incrementally
+compile Dart code.
```

Comment:
Hmm, not really. A static type system allows a compiler to **generate more efficient code**, but it doesn't really make the compilation quicker or more modular.

Reply:
@vsmenon, what do you say?

Reply:
I think you could say "generate more efficient code" and leave it at that. 

For DDC / Web, sound types let us avoid whole program analysis - that's where the quicker / modular comes in.  E.g., if you took the Dart2JS approach without whole program analysis, the generated code would be very slow - possibly too slow even for development.  With Dart 2 / DDC, we can type check modularly and use those types to generate pretty good modularly.

Not sure it's worth going into that level of detail though?